shared/constants/network.ts

```diff
@@ -1034,11 +1034,11 @@ export const CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP = {
  * Ethereum JSON-RPC methods that are known to exist but that we intentionally
  * do not support.
  */
-export const UNSUPPORTED_RPC_METHODS = new Set([
+export const UNSUPPORTED_RPC_METHODS = [
```

Comment:
Nit: Why convert this from a set to an array? Set makes sense to me given that the order is not meaningful, nor would duplicate entries be expected.

Reply:
No good reason. I just generally don't use Sets. Reverted back to Set here https://github.com/MetaMask/metamask-extension/pull/27782/commits/259e36aed7bdd4b7d6163385e2a8ade4d16f3839